hive/src/main/java/io/confluent/connect/storage/hive/HiveSchemaConverter.java

```diff
@@ -93,4 +131,36 @@ public static TypeInfo convertMap(Schema schema) {
   public static TypeInfo convertPrimitive(Schema schema) {
     return TYPE_TO_TYPEINFO.get(schema.type());
   }
+
+  public static TypeInfo convertPrimitiveMaybeLogical(Schema schema) {
+    if (schema.name() == null) {
+      return convertPrimitive(schema);
+    }
+
+    switch (schema.name()) {
+      case Decimal.LOGICAL_NAME:
+        String scale = schema.parameters().get(Decimal.SCALE_FIELD);
+        String precision = schema.parameters().get(CONNECT_AVRO_DECIMAL_PRECISION_PROP);
+        if (precision == null || Integer.parseInt(precision) <= DECIMAL_PRECISION_DEFAULT) {
+          precision = String.valueOf(DECIMAL_PRECISION_DEFAULT);
```

Comment:
Are we risking data loss if `precision == null`?

Reply:
No, we are not. Actually, I don't need to check all this booleans here, I can just make precision = 38 all the time. I've check Hive's serde code, it will do all the logic trying the best to fit decimal data into decimal schema. If the data is too long to fit in, hive will throw SerdeException.